server/api/itemRouter.js

```diff
@@ -0,0 +1,115 @@
+const Item = require('../db/models/item');
+const Category = require('../db/models/category');
+const express = require('express');
+const router = express.Router();
+
+/**
+ * Add an item
+ * 
+ * req.body: {sourceLink: String,
+ *            highlight: String,
+ *            userId: mongoose.ObjectId,
+ *            placesId: id(optional),
+ *            comment: String (optional),}
+ * 
+ * response: {success: boolean,
+ *            message: String}
+ */
+router.post('/addItem', (req, res, next) => {
+  const newItem = req.body;
+  const item = new Item(newItem);
+  let response = {}; 
+  item.save((err) => {
+    if (err) {
+      response.success = false;
+      response.message = 'error adding item';
+      res.json(response);
+      next(err);
+    } else {
+      response.success = true;
+      response.message = 'item added';
+      res.json(response);
+    }
+  });
+});
+
+/**
+ * Get all items
+ * 
+ * req.body: {userId: mongoose.objectId}
+ * 
+ * response: [{sourceLink: String,
+ *            highlight: String,
+ *            placeId: id(optional),
+ *            comments: String (optional),}]
+ */
+router.get('/getItems', (req, res, next) => {
+  Item.find({ userId: req.body.userId }, (err, items) => {
+    if (err) {
+      next(err);
+    }
+    if (items) {
+      res.json(items);
+    }
+  });
+});
+
+/**
+ * Add item to category
+ * 
+ * req.body: {itemId: mongoose.objectId,
+ *            categoryId: mongoose.objectId}
+ * 
+ * response: {success: boolean,
+ *            message: String}
+ */
+router.put('/addItemToCategory', async (req, res, next) => {
+  let response = {}; 
+  try {
+    const itemId = req.body.itemId;
+    const categoryId = req.body.categoryId;
+    await Item.update({ _id: itemId }, { $push: { categoryIds: this.categoryId } }, done).exec();
+    await Category.update({ _id: categoryId }, { $push: { items: this.itemId } }, done).exec();
+    response.success = true;
+    response.message = 'item added to category'; 
+    res.json(response);
+  } catch (error) {
+    response.success = false;
+    response.message = 'error adding item to category'; 
+    res.json(response);
+    next(error);
+  }
+});
+
+/** 
+ * Delete item
+ * 
+ * req.body: {itemId: mongoose.objectId}
+ * 
+ * response: {success: boolean,
+ *            message: String}
+ */
+router.delete('/deleteItem', async (req, res, next) => {
+  let response = {};
+  try { 
+    const itemId = req.body.itemId; 
+    const itemObject = await Item.findOneAndRemove({ _id: itemId}).exec();
+    const itemCategoryIds = itemObject.categoryIds;
+
+    //deleting this itemId from all the categories it belonged to
+    await Promise.all(itemCategoryIds.map(async (categoryId) => {
+      await Category.update({ _id: categoryId}, { $pull: { items: this.itemId} }, done).exec();
+    }))
```

Comment:
Not sure if this works but can you just use pull to go through all documents? I think it would look something like:
```
db.collection.update(
   { },
   { $pull: { items: this.itemId} },
  done).exec();
```
So that you don't have to send multiple requests?

Reply:
Oh yes thank you! That is much smarter. Made this change for deleteItem and deleteCategory